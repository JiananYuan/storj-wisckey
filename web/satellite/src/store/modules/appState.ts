// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { APP_STATE_ACTIONS } from '@/utils/constants/actionNames';
import { AppState } from '@/utils/constants/appStateEnum';

import { APP_STATE_MUTATIONS } from '../mutationConstants';

export const appStateModule = {
    state: {
        // Object that contains all states of views
        appState: {
            fetchState: AppState.LOADING,
            isAddTeamMembersPopupShown: false,
            isNewProjectPopupShown: false,
            isProjectsDropdownShown: false,
            isAccountDropdownShown: false,
            isDeleteProjectPopupShown: false,
            isDeleteAccountPopupShown: false,
            isSortProjectMembersByPopupShown: false,
            isSuccessfulRegistrationShown: false,
            isSuccessfulProjectCreationPopupShown: false,
            isEditProfilePopupShown: false,
            isChangePasswordPopupShown: false,
            isPaymentSelectionShown: false,
            isCreateProjectButtonShown: false,
        },
        satelliteName: '',
    },
    mutations: {
        // Mutation changing add projectMembers members popup visibility
        [APP_STATE_MUTATIONS.TOGGLE_ADD_TEAMMEMBER_POPUP](state: any): void {
            state.appState.isAddTeamMembersPopupShown = !state.appState.isAddTeamMembersPopupShown;
        },

        // Mutation changing new project popup visibility
        [APP_STATE_MUTATIONS.TOGGLE_NEW_PROJECT_POPUP](state: any): void {
            state.appState.isNewProjectPopupShown = !state.appState.isNewProjectPopupShown;
        },

        // Mutation changing project dropdown visibility
        [APP_STATE_MUTATIONS.TOGGLE_PROJECT_DROPDOWN](state: any): void {
            state.appState.isProjectsDropdownShown = !state.appState.isProjectsDropdownShown;
        },

        // Mutation changing account dropdown visibility
        [APP_STATE_MUTATIONS.TOGGLE_ACCOUNT_DROPDOWN](state: any): void {
            state.appState.isAccountDropdownShown = !state.appState.isAccountDropdownShown;
        },

        // Mutation changing delete project popup visibility
        [APP_STATE_MUTATIONS.TOGGLE_DELETE_PROJECT_DROPDOWN](state: any): void {
            state.appState.isDeleteProjectPopupShown = !state.appState.isDeleteProjectPopupShown;
        },
        // Mutation changing delete account popup visibility
        [APP_STATE_MUTATIONS.TOGGLE_DELETE_ACCOUNT_DROPDOWN](state: any): void {
            state.appState.isDeleteAccountPopupShown = !state.appState.isDeleteAccountPopupShown;
        },
        // Mutation changing 'sort project members by' popup visibility.
        [APP_STATE_MUTATIONS.TOGGLE_SORT_PM_BY_DROPDOWN](state: any): void {
            state.appState.isSortProjectMembersByPopupShown = !state.appState.isSortProjectMembersByPopupShown;
        },
        // Mutation changing 'successful registration' area visibility.
        [APP_STATE_MUTATIONS.TOGGLE_SUCCESSFUL_REGISTRATION](state: any): void {
            state.appState.isSuccessfulRegistrationShown = !state.appState.isSuccessfulRegistrationShown;
        },
        // Mutation changing 'successful project creation' popup visibility.
        [APP_STATE_MUTATIONS.TOGGLE_SUCCESSFUL_PROJECT_CREATION_POPUP](state: any): void {
            state.appState.isSuccessfulProjectCreationPopupShown = !state.appState.isSuccessfulProjectCreationPopupShown;
        },
        [APP_STATE_MUTATIONS.TOGGLE_CHANGE_PASSWORD_POPUP](state: any): void {
            state.appState.isChangePasswordPopupShown = !state.appState.isChangePasswordPopupShown;
        },
        [APP_STATE_MUTATIONS.TOGGLE_EDIT_PROFILE_POPUP](state: any): void {
            state.appState.isEditProfilePopupShown = !state.appState.isEditProfilePopupShown;
        },
        [APP_STATE_MUTATIONS.SHOW_SET_DEFAULT_PAYMENT_METHOD_POPUP](state: any, id: string): void {
            state.appState.setDefaultPaymentMethodID = id;
        },
        [APP_STATE_MUTATIONS.SHOW_DELETE_PAYMENT_METHOD_POPUP](state: any, id: string): void {
            state.appState.deletePaymentMethodID = id;
        },
        [APP_STATE_MUTATIONS.SHOW_CREATE_PROJECT_BUTTON](state: any): void {
            state.appState.isCreateProjectButtonShown = true;
        },
        [APP_STATE_MUTATIONS.HIDE_CREATE_PROJECT_BUTTON](state: any): void {
            state.appState.isCreateProjectButtonShown = false;
        },
        // Mutation that closes each popup/dropdown
        [APP_STATE_MUTATIONS.CLOSE_ALL](state: any): void {
            state.appState.isAccountDropdownShown = false;
            state.appState.isProjectsDropdownShown = false;
            state.appState.isSortProjectMembersByPopupShown = false;
        },
        [APP_STATE_MUTATIONS.CHANGE_STATE](state: any, newFetchState: AppState): void {
            state.appState.fetchState = newFetchState;
        },

        // Mutation changing payment selection visibility
        [APP_STATE_MUTATIONS.TOGGLE_PAYMENT_SELECTION](state: any, value: boolean): void {
            state.appState.isPaymentSelectionShown = value;
        },
        [APP_STATE_MUTATIONS.SET_NAME](state: any, satelliteName: string): void {
            state.satelliteName = satelliteName;
        },
    },
    actions: {
        // Commits mutation for changing app popups and dropdowns visibility state
        [APP_STATE_ACTIONS.TOGGLE_TEAM_MEMBERS]: function ({commit, state}: any): void {
            if (!state.appState.isAddTeamMembersPopupShown) {
                commit(APP_STATE_MUTATIONS.CLOSE_ALL);
            }

            commit(APP_STATE_MUTATIONS.TOGGLE_ADD_TEAMMEMBER_POPUP);
        },
        [APP_STATE_ACTIONS.TOGGLE_NEW_PROJ]: function ({commit, state}: any): void {
            if (!state.appState.isNewProjectPopupShown) {
                commit(APP_STATE_MUTATIONS.CLOSE_ALL);
            }

            commit(APP_STATE_MUTATIONS.TOGGLE_NEW_PROJECT_POPUP);
        },
        [APP_STATE_ACTIONS.TOGGLE_PROJECTS]: function ({commit, state}: any): void {
            if (!state.appState.isProjectsDropdownShown) {
                commit(APP_STATE_MUTATIONS.CLOSE_ALL);
            }

            commit(APP_STATE_MUTATIONS.TOGGLE_PROJECT_DROPDOWN);
        },
        [APP_STATE_ACTIONS.TOGGLE_ACCOUNT]: function ({commit, state}: any): void {
            if (!state.appState.isAccountDropdownShown) {
                commit(APP_STATE_MUTATIONS.CLOSE_ALL);
            }

            commit(APP_STATE_MUTATIONS.TOGGLE_ACCOUNT_DROPDOWN);
        },
        [APP_STATE_ACTIONS.TOGGLE_DEL_PROJ]: function ({commit, state}: any): void {
            if (!state.appState.isDeleteProjectPopupShown) {
                commit(APP_STATE_MUTATIONS.CLOSE_ALL);
            }

            commit(APP_STATE_MUTATIONS.TOGGLE_DELETE_PROJECT_DROPDOWN);
        },
        [APP_STATE_ACTIONS.TOGGLE_DEL_ACCOUNT]: function ({commit, state}: any): void {
            if (!state.appState.isDeleteAccountPopupShown) {
                commit(APP_STATE_MUTATIONS.CLOSE_ALL);
            }

            commit(APP_STATE_MUTATIONS.TOGGLE_DELETE_ACCOUNT_DROPDOWN);
        },
        [APP_STATE_ACTIONS.TOGGLE_SORT_PM_BY_DROPDOWN]: function ({commit, state}: any): void {
            if (!state.appState.isSortProjectMembersByPopupShown) {
                commit(APP_STATE_MUTATIONS.CLOSE_ALL);
            }

            commit(APP_STATE_MUTATIONS.TOGGLE_SORT_PM_BY_DROPDOWN);
        },
        [APP_STATE_ACTIONS.TOGGLE_SUCCESSFUL_REGISTRATION]: function ({commit, state}: any): void {
            if (!state.appState.isSuccessfulRegistrationShown) {
                commit(APP_STATE_MUTATIONS.CLOSE_ALL);
            }

            commit(APP_STATE_MUTATIONS.TOGGLE_SUCCESSFUL_REGISTRATION);
        },
        [APP_STATE_ACTIONS.TOGGLE_SUCCESSFUL_PROJECT_CREATION_POPUP]: function ({commit, state}: any): void {
            if (!state.appState.isSuccessfulProjectCreationPopupShown) {
                commit(APP_STATE_MUTATIONS.CLOSE_ALL);
            }

            commit(APP_STATE_MUTATIONS.TOGGLE_SUCCESSFUL_PROJECT_CREATION_POPUP);
        },
        [APP_STATE_ACTIONS.TOGGLE_CHANGE_PASSWORD_POPUP]: function ({commit}: any): void {
            commit(APP_STATE_MUTATIONS.TOGGLE_CHANGE_PASSWORD_POPUP);
        },
        [APP_STATE_ACTIONS.TOGGLE_EDIT_PROFILE_POPUP]: function ({commit}: any): void {
            commit(APP_STATE_MUTATIONS.TOGGLE_EDIT_PROFILE_POPUP);
        },
        [APP_STATE_ACTIONS.SHOW_SET_DEFAULT_PAYMENT_METHOD_POPUP]: function ({commit, state}: any, methodID: string): void {
            if (!state.appState.setDefaultPaymentMethodID) {
                commit(APP_STATE_MUTATIONS.CLOSE_ALL);
            }

            commit(APP_STATE_MUTATIONS.SHOW_SET_DEFAULT_PAYMENT_METHOD_POPUP, methodID);
        },
        [APP_STATE_ACTIONS.SHOW_DELETE_PAYMENT_METHOD_POPUP]: function ({commit, state}: any, methodID: string): void {
            if (!state.appState.deletePaymentMethodID) {
                commit(APP_STATE_MUTATIONS.CLOSE_ALL);
            }

            commit(APP_STATE_MUTATIONS.SHOW_DELETE_PAYMENT_METHOD_POPUP, methodID);
        },
        [APP_STATE_ACTIONS.SHOW_CREATE_PROJECT_BUTTON]: function ({commit}: any): void {
            commit(APP_STATE_MUTATIONS.SHOW_CREATE_PROJECT_BUTTON);
        },
        [APP_STATE_ACTIONS.HIDE_CREATE_PROJECT_BUTTON]: function ({commit}: any): void {
            commit(APP_STATE_MUTATIONS.HIDE_CREATE_PROJECT_BUTTON);
        },
        [APP_STATE_ACTIONS.CLOSE_POPUPS]: function ({commit}: any): void {
            commit(APP_STATE_MUTATIONS.CLOSE_ALL);
        },
        [APP_STATE_ACTIONS.CHANGE_STATE]: function ({commit}: any, newFetchState: AppState): void {
            commit(APP_STATE_MUTATIONS.CHANGE_STATE, newFetchState);
        },
        [APP_STATE_ACTIONS.TOGGLE_PAYMENT_SELECTION]: function ({commit, state}: any, value: boolean): void {
            commit(APP_STATE_MUTATIONS.TOGGLE_PAYMENT_SELECTION, value);
        },
        [APP_STATE_ACTIONS.SET_SATELLITE_NAME]: function ({commit}: any, satelliteName: string): void {
            commit(APP_STATE_MUTATIONS.SET_NAME, satelliteName);
        },
    },
};
