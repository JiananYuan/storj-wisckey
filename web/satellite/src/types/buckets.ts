// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

/**
 * Exposes all bucket-related functionality.
 */
export interface BucketsApi {
    /**
     * Fetch buckets
     *
     * @returns BucketPage
     * @throws Error
     */
    get(projectId: string, before: Date, cursor: BucketCursor): Promise<BucketPage>;
}

/**
 * Bucket class holds info for Bucket entity.
 */
export class Bucket {
    public bucketName: string;
    public storage: number;
    public egress: number;
    public objectCount: number;
    public since: Date;
    public before: Date;

    constructor(bucketName: string = '', storage: number = 0, egress: number = 0, objectCount: number = 0, since: Date = new Date(), before: Date = new Date()) {
        this.bucketName = bucketName;
        this.storage = storage;
        this.egress = egress;
        this.objectCount = objectCount;
        this.since = since;
        this.before = before;
    }

    public formattedBucketName(): string {
        if (this.bucketName.length < 12) {
            return this.bucketName;
        }

        return this.bucketName.slice(0, 12) + '...';
    }
}

/**
 * BucketPage class holds bucket total usages and flag whether more usages available.
 */
export class BucketPage {
    buckets: Bucket[];
    search: string;
    limit: number;
    offset: number;
    pageCount: number;
    currentPage: number;
    totalCount: number;

    constructor(buckets: Bucket[] = [], search: string = '', limit: number = 0, offset: number = 0, pageCount: number = 0, currentPage: number = 0, totalCount: number = 0) {
        this.buckets = buckets;
        this.search = search;
        this.limit = limit;
        this.offset = offset;
        this.pageCount = pageCount;
        this.currentPage = currentPage;
        this.totalCount = totalCount;
    }
}

/**
 * BucketCursor class holds cursor for bucket name and limit.
 */
export class BucketCursor {
    search: string;
    limit: number;
    page: number;

    constructor(search: string = '', limit: number = 0, page: number = 0) {
        this.search = search;
        this.limit = limit;
        this.page = page;
    }
}
