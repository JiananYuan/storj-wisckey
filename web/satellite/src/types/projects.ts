// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

/**
 * Exposes all project-related functionality.
 */
export interface ProjectsApi {
    /**
     * Creates project.
     *
     * @param createProjectModel - contains project information
     * @throws Error
     */
    create(createProjectModel: CreateProjectModel): Promise<Project>;
    /**
     * Fetch projects.
     *
     * @returns Project[]
     * @throws Error
     */
    get(): Promise<Project[]>;
    /**
     * Update project.
     *
     * @param projectId - project ID
     * @param description - project description
     * @returns Project[]
     * @throws Error
     */
    update(projectId: string, description: string): Promise<void>;
    /**
     * Delete project.
     *
     * @param projectId - project ID
     * @throws Error
     */
    delete(projectId: string): Promise<void>;

    /**
     * Get project limits.
     *
     * @param projectId- project ID
     * throws Error
     */
    getLimits(projectId: string): Promise<ProjectLimits>;
}

/**
 * Project is a type, used for creating new project in backend.
 */
export class Project {
    public constructor(
        public id: string = '',
        public name: string = '',
        public description: string = '',
        public createdAt: string = '',
        public ownerId: string = '',
        public isSelected: boolean = false,
    ) {}
}

/**
 * UpdateProjectModel is a type, used for updating project description.
 */
export class UpdateProjectModel {
    public id: string;
    public description: string;

    public constructor(id: string, description: string) {
        this.id = id;
        this.description = description;
    }
}

/**
 * CreateProjectModel is a type, used for creating project.
 */
export class CreateProjectModel {
    public name: string;
    public description: string;
    public ownerId: string;
}

export class ProjectLimits {
    constructor(
        public bandwidthLimit = 0,
        public bandwidthUsed = 0,
        public storageLimit = 0,
        public storageUsed = 0,
    ) {}
}
