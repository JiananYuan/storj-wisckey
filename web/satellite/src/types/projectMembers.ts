// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

/**
 * ProjectMember stores needed info about user info to show it on UI.
 */
import { SortDirection } from '@/types/common';
import { User } from '@/types/users';

export type OnHeaderClickCallback = (sortBy: ProjectMemberOrderBy, sortDirection: SortDirection) => Promise<void>;

export enum ProjectMemberOrderBy {
    NAME = 1,
    EMAIL,
    CREATED_AT,
}

/**
 * Contains values of project members header component state
 * used in ProjectMembersArea and HeaderArea.
 */
export enum ProjectMemberHeaderState {
    DEFAULT = 0,
    /**
     * Used when some project members selected
     */
    ON_SELECT,
}

/**
 * ProjectMembersApi is a graphql implementation of ProjectMembers API.
 * Exposes all ProjectMembers-related functionality
 */
export interface ProjectMembersApi {
    /**
     * Add members to project by user emails.
     *
     * @param projectId
     * @param emails list of project members email to add
     *
     * @throws Error
     */
    add(projectId: string, emails: string[]): Promise<void>;

    /**
     * Deletes ProjectMembers from project by project member emails
     *
     * @param projectId
     * @param emails
     *
     * @throws Error
     */
    delete(projectId: string, emails: string[]): Promise<void>;

    /**
     * Fetch Project Members
     *
     * @param projectId
     * @param cursor
     *
     * @throws Error
     */
    get(projectId: string, cursor: ProjectMemberCursor): Promise<ProjectMembersPage>;
}

/**
 * ProjectMemberCursor is a type, used for paged project members request.
 */
export class ProjectMemberCursor {
    public constructor(
        public search: string = '',
        public limit: number = 6,
        public page: number = 1,
        public order: ProjectMemberOrderBy = ProjectMemberOrderBy.NAME,
        public orderDirection: SortDirection = SortDirection.ASCENDING) {
    }
}

/**
 * ProjectMembersPage is a type, used to describe paged project members list.
 */
export class ProjectMembersPage {
    public constructor(
        public projectMembers: ProjectMember[] = [],
        public search: string = '',
        public order: ProjectMemberOrderBy = ProjectMemberOrderBy.NAME,
        public orderDirection: SortDirection = SortDirection.ASCENDING,
        public limit: number = 6,
        public pageCount: number = 0,
        public currentPage: number = 1,
        public totalCount: number = 0) {
    }
}

/**
 * ProjectMember is a type, used to describe project member.
 */
export class ProjectMember {
    public user: User;

    public joinedAt: string;
    public isSelected: boolean;

    public constructor(fullName: string, shortName: string, email: string, joinedAt: string, id: string = '') {
        this.user = new User(id, fullName, shortName, email);
        this.joinedAt = joinedAt;
        this.isSelected = false;
    }

    public get name(): string {
        return this.user.getFullName();
    }

    public formattedFullName(): string {
        let fullName: string = this.user.getFullName();

        if (fullName.length > 15) {
            fullName = fullName.slice(0, 15) + '...';
        }

        return fullName;
    }

    public formattedEmail(): string {
        let email: string = this.user.email;

        if (email.length > 16) {
            email = this.user.email.slice(0, 13) + '...';
        }

        return email;
    }

    public joinedAtLocal(): string {
        if (!this.joinedAt) return '';

        return new Date(this.joinedAt).toLocaleDateString();
    }
}
