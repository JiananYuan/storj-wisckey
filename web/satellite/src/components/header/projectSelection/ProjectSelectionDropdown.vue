// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

<template>
    <div class="project-selection-choice-container" id="projectDropdown">
        <div class="project-selection-overflow-container">
            <div class="project-selection-overflow-container__project-choice" @click="closeDropdown">
                <div class="project-selection-overflow-container__project-choice__mark-container">
                    <ProjectSelectionIcon
                        class="project-selection-overflow-container__project-choice__mark-container__image"
                    />
                </div>
                <p class="project-selection-overflow-container__project-choice__selected">
                    {{ selectedProject.name }}
                </p>
            </div>
            <div
                class="project-selection-overflow-container__project-choice"
                @click="onProjectSelected(project.id)"
                v-for="project in projects"
                :key="project.id"
            >
                <p class="project-selection-overflow-container__project-choice__unselected">{{ project.name }}</p>
            </div>
        </div>
    </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';

import ProjectSelectionIcon from '@/../static/images/header/projectSelection.svg';

import { BUCKET_ACTIONS } from '@/store/modules/buckets';
import { PAYMENTS_ACTIONS } from '@/store/modules/payments';
import { PROJECTS_ACTIONS } from '@/store/modules/projects';
import { Project } from '@/types/projects';
import {
    API_KEYS_ACTIONS,
    APP_STATE_ACTIONS,
    PM_ACTIONS,
} from '@/utils/constants/actionNames';

@Component({
    components: {
        ProjectSelectionIcon,
    },
})
export default class ProjectSelectionDropdown extends Vue {
    private FIRST_PAGE = 1;

    /**
     * Fetches all project related information.
     * @param projectID
     */
    public async onProjectSelected(projectID: string): Promise<void> {
        await this.$store.dispatch(PROJECTS_ACTIONS.SELECT, projectID);
        await this.$store.dispatch(PM_ACTIONS.SET_SEARCH_QUERY, '');
        this.closeDropdown();

        try {
            await this.$store.dispatch(PAYMENTS_ACTIONS.GET_PROJECT_USAGE_AND_CHARGES_CURRENT_ROLLUP);
        } catch (error) {
            await this.$notify.error(`Unable to fetch project usage. ${error.message}`);
        }

        try {
            await this.$store.dispatch(PM_ACTIONS.FETCH, this.FIRST_PAGE);
        } catch (error) {
            await this.$notify.error(`Unable to fetch project members. ${error.message}`);
        }

        try {
            await this.$store.dispatch(API_KEYS_ACTIONS.FETCH, this.FIRST_PAGE);
        } catch (error) {
            await this.$notify.error(`Unable to fetch api keys. ${error.message}`);
        }

        try {
            await this.$store.dispatch(BUCKET_ACTIONS.FETCH, this.FIRST_PAGE);
        } catch (error) {
            await this.$notify.error('Unable to fetch buckets: ' + error.message);
        }

        try {
            await this.$store.dispatch(PROJECTS_ACTIONS.GET_LIMITS, this.$store.getters.selectedProject.id);
        } catch (error) {
            await this.$notify.error(`Unable to fetch project limits. ${error.message}`);
        }
    }

    /**
     * Returns projects list from store.
     */
    public get projects(): Project[] {
        return this.$store.getters.projectsWithoutSelected;
    }

    /**
     * Returns selected project from store.
     */
    public get selectedProject(): Project {
        return this.$store.getters.selectedProject;
    }

    /**
     * Closes dropdown.
     */
    public closeDropdown(): void {
        this.$store.dispatch(APP_STATE_ACTIONS.TOGGLE_PROJECTS);
    }
}
</script>

<style scoped lang="scss">
    ::-webkit-scrollbar,
    ::-webkit-scrollbar-track,
    ::-webkit-scrollbar-thumb {
        margin: 0;
        width: 0;
    }

    .project-selection-choice-container {
        position: absolute;
        top: 65px;
        left: -5px;
        background-color: #fff;
        z-index: 1120;
        border: 1px solid #c5cbdb;
        box-shadow: 0 8px 34px rgba(161, 173, 185, 0.41);
        border-radius: 6px;
    }

    .project-selection-overflow-container {
        position: relative;
        min-width: 226px;
        width: auto;
        overflow-y: scroll;
        height: auto;
        max-height: 240px;
        background-color: #fff;
        font-family: 'font_regular', sans-serif;
        border-radius: 6px;

        &__project-choice {
            display: flex;
            align-items: center;
            justify-content: flex-start;
            padding: 0 25px;

            &__selected,
            &__unselected {
                margin: 12px 0;
                font-size: 14px;
                line-height: 20px;
                color: #7e8b9c;
                word-break: break-all;
            }

            &__selected {
                font-family: 'font_bold', sans-serif;
                color: #494949;
            }

            &:hover {
                background-color: #f2f2f6;

                .project-selection-overflow-container__project-choice__unselected {
                    color: #354049;
                }
            }

            &__mark-container {
                width: 10px;
                margin-right: 12px;

                &__image {
                    object-fit: cover;
                }
            }
        }
    }

    .arrow {
        padding-right: 25px;
    }

    @media screen and (max-width: 1280px) {

        .project-selection-choice-container {
            top: 50px;
        }
    }
</style>
