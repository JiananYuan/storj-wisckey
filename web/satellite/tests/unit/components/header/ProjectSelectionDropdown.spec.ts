// Copyright (C) 2020 Storj Labs, Inc.
// See LICENSE for copying information.

import Vuex from 'vuex';

import ProjectSelectionDropdown from '@/components/header/projectSelection/ProjectSelectionDropdown.vue';

import { makeProjectsModule, PROJECTS_MUTATIONS } from '@/store/modules/projects';
import { Project } from '@/types/projects';
import { createLocalVue, shallowMount } from '@vue/test-utils';

import { ProjectsApiMock } from '../../mock/api/projects';

const localVue = createLocalVue();
localVue.use(Vuex);

const projectsApi = new ProjectsApiMock();
const projectsModule = makeProjectsModule(projectsApi);
const project1 = new Project('testId1', 'testName1', '');
const project2 = new Project('testId2', 'testName2', '');

const store = new Vuex.Store({ modules: { projectsModule }});

describe('ProjectSelectionDropdown', () => {
    it('renders correctly', () => {
        store.commit(PROJECTS_MUTATIONS.SET_PROJECTS, [project1, project2]);
        store.commit(PROJECTS_MUTATIONS.SELECT_PROJECT, project1.id);

        const wrapper = shallowMount(ProjectSelectionDropdown, {
            store,
            localVue,
        });

        expect(wrapper).toMatchSnapshot();
    });
});
