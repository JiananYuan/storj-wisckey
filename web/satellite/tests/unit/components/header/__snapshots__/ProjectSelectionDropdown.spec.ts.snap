// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectSelectionDropdown renders correctly 1`] = `
<div id="projectDropdown" class="project-selection-choice-container">
  <div class="project-selection-overflow-container">
    <div class="project-selection-overflow-container__project-choice">
      <div class="project-selection-overflow-container__project-choice__mark-container">
        <projectselectionicon-stub class="project-selection-overflow-container__project-choice__mark-container__image"></projectselectionicon-stub>
      </div>
      <p class="project-selection-overflow-container__project-choice__selected">
        testName1
      </p>
    </div>
    <div class="project-selection-overflow-container__project-choice">
      <p class="project-selection-overflow-container__project-choice__unselected">testName2</p>
    </div>
  </div>
</div>
`;
